### PR TITLE
Customize: Fix adding a widget when the widget block editor is disabled

### PR DESCRIPTION
**Pull request by noisysocks**
Trac ticket: https://core.trac.wordpress.org/ticket/53479

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by adamziel — APPROVED** [review]
Great job adding all the tests!

File: tests/phpunit/tests/customize/widgets.php [hunk 1]
```diff
@@ -401,6 +401,71 @@ function test_sanitize_widget_js_instance() {
...
+
+		$unsanitized_from_js = $this->manager->widgets->sanitize_widget_instance( $sanitized_for_js );
+		$this->assertSame( $unsanitized_from_js, $block_instance );
+	}
+
+	/**
+	 * There should NOT be a 'raw_instance' key when the block editor is enabled
+	 * but the widget does not support them because `show_instance_in_rest` on
+	 * the widget is set to false.
+	 */
```
Comment (hellofromtonya) [inline tests/phpunit/tests/customize/widgets.php:432]:
```suggestion
	 *
	 * @group 53489
	 */
```

File: tests/phpunit/tests/customize/widgets.php [hunk 2]
```diff
@@ -401,6 +401,71 @@ function test_sanitize_widget_js_instance() {
 		$this->assertSame( $unsanitized_from_js, $new_categories_instance );
 	}
 
+	/**
+	 * There should be a 'raw_instance' key when the block editor is enabled and
+	 * the widget supports them via `show_instance_in_rest`.
+	 */
```
Comment (hellofromtonya) [inline tests/phpunit/tests/customize/widgets.php:407]:
```suggestion
	 *
	 * @group 53479
	 */
```

**Review by hellofromtonya — APPROVED** [review]
- Commented out the `return $value` change, ran the tests, and the `test_sanitize_widget_instance_empty_instance` test failed (as it should) ✅ 
- Uncommented the `return $value` change, ran the tests, and the `test_sanitize_widget_instance_empty_instance` test passed with the change (as it should) ✅

**Pull request closed without merging**